### DISCUSSION TITLE
URL Slug is not matching slug

### DISCUSSION THREAD

**Discussion by devsrealm** [pimcore/pimcore#15779] [Q&A / Support] [answered]
I don't know what is going on with my URL Slug Field, no matter what I tried, it doesn't match the slug, the [documentation ](https://pimcore.com/docs/platform/Pimcore/Objects/Object_Classes/Data_Types/Others#url-slug)is not clear on that as well, though I tried, both:

- `App\Controller\ProductlistController:action` and 
- `\App\Controller\ProductlistController::action` 

notice that double `:` in the second version, nothing works.

My controller looks like so:

```
 public function action(ShopProduct $ShopProduct , UrlSlug $urlSlug): Response
 {

 }
```

Can someone please help me with this?

**Comment by devsrealm** [comment]
I am using Version: v10.5.19

**Comment by twin-elements** [comment, marked as answer]
Hi. 
I agree, the documentation is completely unclear.

In the model, in the slug field, add the `App\Controller\YourController::slugAction`

![image](https://github.com/pimcore/pimcore/assets/17727266/8d4e1c09-c4c5-4a28-8ca7-c3c8a896aebb)

slugAction should direct to the target method in your controller

![image](https://github.com/pimcore/pimcore/assets/17727266/dd8c260a-0654-4435-8e50-7e89575b9cf9)

**Comment by devsrealm** [comment]
@twin-elements The mistake I made was I was prefixing the URL with `/{_locale}/shop`, is there any way I can make it take that into account though?

**Reply by twin-elements** [reply]
Show me what your method looks like in the controller please.

**Reply by devsrealm** [reply]
The method is currently a complete replica of what you suggested in the image.

Something like below:

```
public function slugAction(Request $request, Foo $object, UrlSlug $urlSlug) 
{
        
}
```

I couldn't get the object to work though (can't resolve it yet), it is complaining about autowire or something like that, at least, it is matching the slug for now, so, that is a good first step.

Between, I can use it like so:

`example.com/slug`

I would like to be able to use it like so:

`example.com/{_locale}/shop/slug`

For example `example.com/en/shop/slug ` without having to add the `en/shop` part to the slug value, I hope you understand.

**Reply by devsrealm** [reply]
An idea I am thinking through is by generating the slug myself, but down the line, it looks unmaintainable.

**Reply by VincentBibauw** [reply]
I looking into the same thing, but I haven't found a good solution yet.

**Reply by twin-elements** [reply]
You must pass the necessary parameters to the controller in the slugAction method 

![image](https://github.com/pimcore/pimcore/assets/17727266/e98cfa0e-d4f4-4df0-940d-02a363377180)

![image](https://github.com/pimcore/pimcore/assets/17727266/e8a6da9e-8680-4546-9492-71684a892be6)

Then everything should work

**Comment by twin-elements** [comment]
> Metoda jest obecnie pełną repliką tego, co zasugerowałeś na obrazku.
> 
> Coś jak poniżej:
> 
> ```
> public function slugAction(Request $request, Foo $object, UrlSlug $urlSlug) 
> {
>         
> }
> ```
> 
> Nie udało mi się jednak zmusić obiektu do działania (jeszcze nie mogę go rozwiązać), narzeka na autowire lub coś w tym stylu, przynajmniej na razie pasuje do ślimaka, więc jest to dobry pierwszy krok.
> 
> Pomiędzy mogę go używać w ten sposób:
> 
> `example.com/slug`
> 
> Chciałbym móc go używać w ten sposób:
> 
> `example.com/{_locale}/shop/slug`
> 
> Mam nadzieję, że rozumiesz, na przykład `example.com/en/shop/slug `bez konieczności dodawania tej części do wartości ślimaka.`en/shop`

Then you do not use a **slug**, but a link generator
